Add support for healthcheck in compose to swarm

vdemeester opened the pull request:
Fixes #28520

Adds support for `healtcheck` in the composefile v3 format, used it `docker stack deploy --compose-file` 🐮.

```yaml
version: "3"

services:
  foo:
    image: busybox
    command: top
    healthcheck:
      test: cat /etc/password
      interval: 10s
      timeout: 2s
      retries: 3
```

/cc @dnephin @aanand @icecrime 

This bumps [compose-file](https://github.com/aanand/compose-file) to a70cb6ea5c403c383b5117dcd1dea78880a47824.

🐸

cpuguy83 commented on cli/command/stack/deploy.go:
```diff
@@ -520,6 +528,45 @@ func convertService(
+	)
+	// FIXME(vdemeester) decide what mean NONE for healthcheck
+	if len(healthcheck.Command) == 0 || healthcheck.Command[0] == "none" {
+		test = []string{"NONE"}
```
I'd not use a magic value.
Something like `DisableHealthCheck: true` seems good to me.
Then someone can just add or remove this to the existing config to enable/disable.

vdemeester replied:
/cc @aanand @dnephin

dnephin replied:
hmm, ya I guess we could do:

```yaml
healthcheck:
  disable: true
```

And then error if any other fields are set at the same time as `disable: true` .

vdemeester replied:
sounds good to me 👼

cpuguy83 replied:
I don't think we'd want to error, would we? Then people can't turn it on/off quickly.

dnephin replied:
Yes, this needs to be an error:

```yaml
healthcheck:
  disable: true
  command: foo
```

That config doesn't make any sense. It doesn't make any more time to comment out the `command` line.

vdemeester replied:
I'll do a PR in compose-file for that then (/cc @aanand for the `docker/compose` part)

vdemeester committed:
Update compose-file to bc280351b9589b05c6da031d9f93c5c5d22db922

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester committed:
Add support for healthcheck in composefile v3

`docker stack deploy` now supports a composefile v3 format that have a
healthcheck.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
@aanand @dnephin updated with latest changes (`commant` -> `test`, `disable`, …)

dnephin committed:
Default parallelism to 1.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

dnephin commented on cli/command/stack/deploy.go:
```diff
@@ -531,6 +539,47 @@ func convertExtraHosts(extraHosts map[string]string) []string {
+		if err != nil {
+			return nil, err
+		}
+	}
```
I guess these could be `time.Duration` in the compose-file/types, but we can fix that alter.

vdemeester replied:
Ah good point :stuck_out_tongue:

dnephin reviewed (approved):
LGTM

vieux commented:
LGTM
